rules.mk

```diff
@@ -110,7 +110,7 @@ KERNCFLAGS+=    -mno-sse \
 		-mno-sse2
 endif
 KERNCFLAGS+=	-fno-omit-frame-pointer
-KERNLDFLAGS=	--gc-sections -n -L $(OUTDIR)/klib
+KERNLDFLAGS=	--gc-sections -z max-page-size=4096 -L $(OUTDIR)/klib
```

Comment:
Just curious what this is for - were you seeing issues with larger page sizes being used for aarch64?

Reply:
The `-n` linker flag turns off page alignment of ELF sections, and this was causing issues because the boot code that maps the kernel ELF binary requires the file offset of all sections to have the same page offset as the base virtual address. But without the `-n` flag, the linker (on x86_64) was aligning ELF sections to 2MB boundaries (which was increasing the kernel binary file size considerably), so I added the `-z max-page-size=4096` flag.